tiledb/sm/global_state/tbb_state.cc

```diff
@@ -48,9 +50,16 @@ static std::unique_ptr<tbb::task_scheduler_init> tbb_scheduler_;
 /** The number of TBB threads the scheduler was configured with **/
 static int tbb_nthreads_;
 
-Status init_tbb(tiledb::sm::Config* config) {
-  int nthreads = config ? config->sm_params().num_tbb_threads_ :
-                          constants::num_tbb_threads;
+Status init_tbb(const Config* config) {
+  int nthreads;
+  if (!config) {
+    nthreads = atoi(constants::sm_num_tbb_threads.c_str());
```

Comment:
`atoi` can throw an exception, which should probably be caught here since this is a part of global state initialization (so there may be no way to catch it cleanly at the call site)

Reply:
I don't think `atoi` throws an exception: http://www.cplusplus.com/reference/cstdlib/atoi/. I will change that to `strtol` though that returns `0` on error. We handle the error case when`nthreads <1` a few lines later in that function.